### DISCUSSION TITLE
how to get paired joint importance using subsample

### DISCUSSION THREAD

**Discussion by dimstatical** [kogalur/randomForestSRC#225] [Q&A] [answered]
Thank you for your great work with the randomforestSRC package. I wonder if the subsample function could calculate the joint importance of selected variables like the vimp function does. 
#For example,
library(randomForestSRC)
airq.obj<-rfsrc(Ozone~.,data=airquality)
v<-vimp(airq.obj,c("Wind","Temp"),joint = TRUE) #return the joint importance of "Wind" and "Temp"

o<-subsample(airq.obj,c("Wind","Temp"),joint = TRUE)
##this returns: Error in max - min : non-numeric argument to binary operator
Hope you could give me some suggestions, Thank you!

**Comment by ishwaran** [comment, marked as answer]
Unfortunately the current function cannot do this.  But this is a very good suggestion and I have added it to the new build which we will be posting in a few days to CRAN.  

Here's an example of what the call will look like:

```
o <- rfsrc(Ozone ~ ., airquality)
oo <- subsample(o,  joint = TRUE, xvar.names = c("Day", "Month"))
plot.subsample(oo)
```

**Reply by dimstatical** [reply]
Thank you for the quick reply! I am looking foward it. 
I have to say that similar questions existed in the get.mv.vimp function for multivariate regression. Specifically in my interests, the get.mv.vimp function favors in obtain standardized importance but could not calculate them jointly, whereas the vimp function can obtain joint importance of selected predictors but can not get  standardized importance. I wonder if there will be ways to improve the situation since I am not quite familiar with the source C code.
In addition, I also donot understand how the standardize works, like the tutorial said: 'For regression families, VIMP is standardized by dividing by the variance and then multipled by 100'. could you kindly let me know how the variance is calculated? using var()? Take the mtcars for a example, a constant of 36.3241 (not the var or sd of mpg importance) was used to divide the vimp for mpg. How was this number obtained?
`mtcars2 <- mtcars`
`mtcars2$cyl <- factor(mtcars2$cyl)`
`mtcars2$carb <- factor(mtcars2$carb, ordered = TRUE)`
`mv.obj <- rfsrc(cbind(carb, mpg, cyl) ~., data = mtcars2, importance = TRUE)`
`v1<-get.mv.vimp( mv.obj)`
         carb        mpg         cyl
disp 0.102313172 25.5203215 0.377262097
hp   0.195611559 16.7714045 0.231893817
drat 0.028145161  1.8829295 0.013083333
wt   0.022723118 13.4938033 0.111102151
qsec 0.040158602  0.9988012 0.033575269
vs   0.044053763  1.5440238 0.081055108
am   0.004435484  0.2079411 0.001834677
gear 0.025245968  1.0225569 0.006228495
` v2<-get.mv.vimp( mv.obj,standardize = T)`
carb         mpg         cyl
disp 0.102313172 0.702572658 0.377262097
hp   0.195611559 0.461715588 0.231893817
drat 0.028145161 0.051836917 0.013083333
wt   0.022723118 0.371483457 0.111102151
qsec 0.040158602 0.027496926 0.033575269
vs   0.044053763 0.042506868 0.081055108
am   0.004435484 0.005724605 0.001834677
gear 0.025245968 0.028150921 0.006228495
`v1/v2`
carb     mpg cyl
disp    1 36.3241   1
hp      1 36.3241   1
drat    1 36.3241   1
wt      1 36.3241   1
qsec    1 36.3241   1
vs      1 36.3241   1
am      1 36.3241   1
gear    1 36.3241   1
Thank you again for the great work of your group!

**Reply by ishwaran** [reply]
`get.mv.vimp()` is an extraction function.  If there is VIMP in the object then it extracts the VIMP (and it can standardize or not standardize this value).   It ****does not** calculate** VIMP and if you want joint VIMP for example, then you should use the function `vimp()`.   

Standardization means that the VIMP is divided by the variance of the y response for regression.  Standardization only applies for regression outcomes.  In your case you have a multivariate family, but only one outcome is continuous and therefore only that variable's VIMP will be standardized.

```
mtcars2 <- mtcars
mtcars2$cyl <- factor(mtcars2$cyl)
mtcars2$carb <- factor(mtcars2$carb, ordered = TRUE)
o <- rfsrc(cbind(carb, mpg, cyl) ~., data = mtcars2)

v <- vimp(o, joint = TRUE)

print(get.mv.vimp(v))
print(get.mv.vimp(v, standardize = TRUE))

```

**Reply by dimstatical** [reply]
OK, I get it now. Thank you again and have a nice weekend!
